### ISSUE TITLE
[sign_in_with_apple] Exception when AuthorizationScopes is set to email only

### ISSUE THREAD

**Issue by peterstojanowski** [aboutyou/dart_packages#115]
Hello,
I noticed that the `SignInWithApple.getAppleIDCredential` throws an exception when I sign in for the first time and don't select the fullName authorisation scope.

Here's what I'm seeing in logs:
```
Exception has occurred.
NoSuchMethodError (NoSuchMethodError: The method '[]' was called on null.
Receiver: null
Tried calling: []("firstName"))
```

Looks like it this is the problem (authorisation_credential.dart):
```
return AuthorizationCredentialAppleID(
    authorizationCode: deeplink.queryParameters['code'],
    email: user != null ? user['email'] as String : null,
    givenName: user != null ? user['name']['firstName'] as String : null, // here
    familyName: user != null ? user['name']['lastName'] as String : null, // and here
    userIdentifier: null,
    identityToken: deeplink.queryParameters['id_token'],
    state: deeplink.queryParameters['state'],
  );
```

**Comment by HenriBeck** [comment]
Hey @peterstojanowski 

Thanks for reporting the issue. I found the root problem and will provide a fix tomorrow for it

**Comment by HenriBeck** [comment]
@peterstojanowski I just released version 2.5.2 which has this bug fixed.
